lib/gc.cpp

```diff
@@ -81,6 +81,34 @@ static void maybe_initialize_gc() {
     }
 }
 
+#ifdef __GLIBC__
+#define GNOEXCEPT __THROW
+#else
+#define GNOEXCEPT
+#endif
+void *operator new(std::size_t size) __attribute__((used));
```

Comment:
Please add a comment why these prototypes are necessary.

Reply:
```suggestion
[[gnu::used]] void *operator new(std::size_t size);
```

Would this work? This syntax is safer as other compilers will just ignore it, unlike the `__attribute__` syntax.

Reply:
Also, the exception declarations don't still don't match those in `<new>` which could be a problem.

Reply:
It may be that we don't need to be messing with `new` and friends at all.  Let me check to see.  I'm going to try to reproduce the original issue with `p4c` master since the problem originally presented in an internal branch.